app/nexus/internal/state/persist/persist.go

```diff
@@ -37,11 +37,11 @@ var (
 // The function does not return any errors since it handles them internally
 // through logging. Cache failures are non-fatal as the KV store is considered
 // the authoritative data source.
-func AsyncPersistSecret(kv *store.KV, path string) {
+func AsyncPersistSecret(kv *store.KV, path string) error {
 	be := Backend()
 
 	// Get the full secret for caching
-	secret := kv.GetRawSecret(path)
+	secret, err := kv.GetRawSecret(path)
```

Comment:
Better to log the error here.

It could be possible that before we get secret to persist, someone else might have already deleted it because the following mutex does not lock over the async operation (and that's intentional).

```
	kvMu.Lock()
	kv.Put(path, values)
	kvMu.Unlock()

	persist.AsyncPersistSecret(kv, path)
```

So, not finding a secret is not an error; but more like a "warning" to be logged.

Reply:
I left it as an info.